python/ray/autoscaler/v2/scheduler.py

```diff
@@ -157,27 +175,47 @@ def new(
             return SchedulingNode(
                 node_type=instance.im_instance.instance_type,
                 total_resources=dict(instance.ray_node.total_resources),
-                # Use ray node's available resources.
-                available_resources=dict(instance.ray_node.available_resources),
+                # Available resources for scheduling requests of different
+                # sources.
+                available_resources={
+                    # Demand are fulfilled with available resources.
+                    ResourceRequestSource.PENDING_DEMAND: dict(
+                        instance.ray_node.available_resources
+                    ),
+                    # Cluster resource constraints are fulfilled with total
+                    # resources of a node.
+                    ResourceRequestSource.CLUSTER_RESOURCE_CONSTRAINT: dict(
+                        instance.ray_node.total_resources
+                    ),
+                },
```

Comment:
Would this be better?

```
class SchedulingNode(total_resources, available_resources):
    def get_resources_for_scheduling(resource_request_source):
       if source == pending_demand:
           return available_resources
       else:
           return total_resources
 ```

Reply:
this would work, one caveat is we also need the total resources for computing resource utilization post scheduling. So we need:
1. available resources for demand
2. available resources for cluster resource constraint
3. total resources

for a single node, and these 3 would be of different values at one point.